test/com/google/javascript/jscomp/integration/ProductionCoverageInstrumentationPassIntegrationTest.java

```diff
@@ -101,26 +110,195 @@ public void testNoTranspilationFunctionInstrumentation() {
+               getInstrumentCodeLine("C", 1, 0), ";",
+            "  console.log('Hello'); ",
+            "} else {",
+               getInstrumentCodeLine("E", 1, 0), ";",
+            "}");
+
+    String[] expectedArr = {instrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  @Test
+  public void testOrInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source =
+        lines("tempObj.a || tempObj.b;");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines(
+            "tempObj.a || (",
+            getInstrumentCodeLine("C", 1, 13), ", tempObj.b)");
+
+    String[] expectedArr = {instrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  @Test
+  public void testAndInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source =
+        lines("tempObj.a && tempObj.b;");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines(
+            "tempObj.a && (",
+            getInstrumentCodeLine("C", 1, 13), ", tempObj.b)");
+
+    String[] expectedArr = {instrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  @Test
+  public void testCoalesceInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    options.setLanguageIn(LanguageMode.ECMASCRIPT_2020);
+    options.setLanguageOut(LanguageMode.NO_TRANSPILE);
+
+    String source = lines("someObj ?? 24");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines(
+            "someObj ?? (",
+            getInstrumentCodeLine("C", 1, 11), ", 24)");
+
+    String[] expectedArr = {noTranspilationInstrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  @Test
+  public void testTernaryInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source = lines("flag ? foo() : fooBar()");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines(
+            "flag ? (",
+            getInstrumentCodeLine("C", 1, 7), ", foo()) :",
+            "(",
+            getInstrumentCodeLine("C", 1, 15), ", fooBar())");
+
+    String[] expectedArr = {instrumentCodeExpected, expected};
+    test(options, sourceArr, expectedArr);
+  }
+
+  @Test
+  public void testForLoopInstrumentation() {
+    CompilerOptions options = createCompilerOptions();
+
+    String source =
+        lines(
+            "for(var i = 0 ; i < 10 ; ++i) {",
+            "   console.log('*');",
+            "}");
+
+    String[] sourceArr = {instrumentCodeSource, source};
+
+    String expected =
+        lines(
+            "for(var i = 0 ; i < 10 ; ++i) {",
+                getInstrumentCodeLine("C", 1, 0), ";",
+            "   console.log('*');",
+            "}",
+            getInstrumentCodeLine("E", 1, 0), ";");
```

Comment:
Tbh I'm not sure what benefit this case adds. In `if` or `switch` statements you use default statements to detect when main condition never runs. While here there is no default case for loop. So if this line runs - it doesn't give any useful info about whether loop ran or not.

Reply:
I don't really see that much benefit either. We can also add it back after if a use case presents itself.